library/core/src/mem/maybe_uninit.rs

```diff
@@ -367,8 +371,7 @@ impl<T> MaybeUninit<T> {
         [const { MaybeUninit::uninit() }; N]
     }
 
-    /// Creates a new `MaybeUninit<T>` in an uninitialized state, with the memory being
-    /// filled with `0` bytes. It depends on `T` whether that already makes for
+    /// Creates a new zero-filled `MaybeUninit<T>`. It depends on `T` whether that already makes for
     /// proper initialization. For example, `MaybeUninit<usize>::zeroed()` is initialized,
```

Comment:
What about this one?

Reply:
The old text is even trying to claim an uninitialized state, even though it depends on `T`.

Reply:
Hm, yeah I could see how this could be confusing.